### PR TITLE
[bugfix] List inherited pipeline hooks with the `-L` option

### LINKED ISSUE #1566: Inherited pipeline hooks are not listed with the `-L` option
For example, the `cscs-checks/prgenv/cuda/cuda_samples.py` tests inherit their hook from the base `CudaSamples`, but this does not show up when listing the concrete tests:

```
[List of matched checks]
- CudaDeviceQueryCheck:
    Dependencies:
      <none>

    Description:
      CUDA deviceQuery test.

    Environment modules:
      cudatoolkit

    Location:
      /users/karakasv/Devel/reframe/cscs-checks/prgenv/cuda/cuda_samples.py

    Maintainers:
      JO

    Node allocation:
      standard (1 task(s))

    Pipeline hooks:

    Tags:
      external_resosurces, production

    Valid environments:
      PrgEnv-cray, PrgEnv-gnu, PrgEnv-pgi

    Valid systems:
      daint:gpu, dom:gpu, tiger:gpu, arolla:cn, tsa:cn, ault:amdv100, ault:intelv100

Found 1 check(s)
```

### PR DESCRIPTION
**Pull request by ekouts**
Closes #1566.

### CODE REVIEW

File: reframe/frontend/cli.py [hunk 1]
```diff
@@ -63,6 +63,13 @@ def fmt_deps():
     else:
         node_alloc_scheme = f'flexible (minimum {-check.num_tasks} task(s))'
 
+    hooks = {}
+    for c in inspect.getmro(type(check)):
+        if hasattr(c, '_rfm_pipeline_hooks'):
+            for k, v in c._rfm_pipeline_hooks.items():
+                hooks.setdefault(k, [])
+                hooks[k].extend(v)
```
Comment (vkarak) [inline reframe/frontend/cli.py:71]: This is too esoteric to be exposed here. I suggest having a function at the class level the returns all the hooks.

**Review by vkarak — CHANGES_REQUESTED** [review]

**Review by vkarak — CHANGES_REQUESTED** [review]
This implementation is very close to this one:

https://github.com/eth-cscs/reframe/blob/67e66aad96ae751186c2efa58d319f34b6a394ba/reframe/core/pipeline.py#L84

I think we should have a class method that will iterate over the hooks walking through the `mro`. This will be used also in the place shown above.

**Comment by ekouts** [comment]
@vkarak In the listing of the tests ith `-L` do we want to print disabled hooks as well?

**Comment by vkarak** [comment]
> @vkarak In the listing of the tests with -L do we want to print disabled hooks as well?

I thought the same thing yesterday, but I don't have a strong opinion. For the moment I would say, whichever is the most straightforward to implement.

File: reframe/core/pipeline.py [hunk 2]
```diff
@@ -148,6 +148,17 @@ def disable_hook(cls, hook_name):
         '''
         cls._rfm_disabled_hooks.add(hook_name)
 
+    @classmethod
+    def hooks(cls):
```
Comment (vkarak) [inline reframe/core/pipeline.py:152]:
```suggestion
    def pipeline_hooks(cls):
```

File: reframe/frontend/cli.py [hunk 3]
```diff
@@ -71,7 +71,7 @@ def fmt_deps():
         'Node allocation': node_alloc_scheme,
         'Pipeline hooks': {
             k: fmt_list(fn.__name__ for fn in v)
-            for k, v in type(check)._rfm_pipeline_hooks.items()
+            for k, v in type(check).hooks().items()
```
Comment (vkarak) [inline reframe/frontend/cli.py:74]: This works as well.

```suggestion
            for k, v in check.pipeline_hooks().items()
```

**Review by vkarak — APPROVED** [review]

**Comment by vkarak** [comment]
@jenkins-cscs retry dom

**Pull request merged**